feature/notification/src/main/java/io/github/droidkaigi/confsched2019/notification/NotificationUtil.kt

```diff
@@ -0,0 +1,100 @@
+import android.content.Context
+import android.os.Build
+import androidx.annotation.DrawableRes
+import androidx.annotation.RequiresApi
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationManagerCompat
+
+fun Context.notificationBuilder(
+    channelInfo: NotificationChannelInfo
+): NotificationCompat.Builder {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+        createDefaultNotificationChannel(
+            this,
+            channelInfo
+        )
+    }
+
+    return NotificationCompat.Builder(this, channelInfo.channelId)
+}
+
+@RequiresApi(Build.VERSION_CODES.O)
+internal fun createDefaultNotificationChannel(
+    context: Context,
+    notificationChannelInfo: NotificationChannelInfo
+) {
+    val notificationManager = context.getSystemService(NotificationManager::class.java)
+    val channel = NotificationChannel(
+        notificationChannelInfo.channelId,
+        notificationChannelInfo.channelName(context),
+        NotificationManager.IMPORTANCE_DEFAULT
+    )
+    notificationManager.createNotificationChannel(channel)
+}
+
+object NotificationUtil {
+    fun showNotification(
+        context: Context,
+        title: String,
+        text: String,
+        pendingIntent: PendingIntent?,
+        channelInfo: NotificationChannelInfo = NotificationChannelInfo.DEFAULT,
+        @DrawableRes iconRes: Int = R.drawable.ic_notification,
+        builder: NotificationCompat.Builder.() -> Unit = {}
+    ) {
+        val notificationBuilder = context.notificationBuilder(channelInfo).apply {
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+                showBundleNotification(
+                    context,
+                    title,
+                    channelInfo,
+                    iconRes
+                )
+                setGroup(channelInfo.channelId)
+            } else {
+                setContentTitle(title)
+            }
+            setContentText(text)
+            setStyle(
+                NotificationCompat.BigTextStyle()
+                    .bigText(text)
+            )
+
+            if (pendingIntent != null) {
+                setContentIntent(pendingIntent)
+            }
+            setAutoCancel(true)
+            setSmallIcon(iconRes)
+        }.apply(builder)
+
+        val notificationManagerCompat = NotificationManagerCompat.from(context)
+        notificationManagerCompat.notify(text.hashCode(), notificationBuilder.build())
+    }
+
+    @RequiresApi(Build.VERSION_CODES.N)
+    private fun showBundleNotification(
+        context: Context,
+        title: String,
+        channelInfo: NotificationChannelInfo,
+        @DrawableRes
+        iconRes: Int
+    ) {
+        val notificationManager = context.getSystemService(NotificationManager::class.java)
+        val notification = context.notificationBuilder(channelInfo)
+            .setStyle(
+                NotificationCompat.BigTextStyle()
+                    .setSummaryText(title)
```

Comment:
I checked on Android 8 of the emulator and Android 9 of the Essential Phone, but it seems I can not see the title when there is one notification. Is it supposed to be setContentTitle?
![image](https://user-images.githubusercontent.com/1386930/51263775-4adb9880-19f8-11e9-9d20-6298129f32ee.png)

Reply:
oh, it seems true... this method is just migrated from the existing code though, i need to fix. Thanks always. 👀

Reply:
> from the existing code

😭